syspage.h

```diff
@@ -19,112 +19,122 @@
 #include <hal/hal.h>
 
 
-#define SIZE_APP_NAME 31
-#define SIZE_MAP_NAME 7
+/* syspage_addProg bitflags */
+enum { flagSyspageExec = 0x01 };
 
 
-/* TODO: Make it compatible with Phoenix-RTOS kernel;
- *       take into account map's attributes while data is written to them */
 enum { mAttrRead = 0x01, mAttrWrite = 0x02, mAttrExec = 0x04, mAttrShareable = 0x08,
 	   mAttrCacheable = 0x10, mAttrBufferable = 0x20 };
 
 
-/* syspage_addProg bitflags */
-enum { flagSyspageExec = 0x01 };
+#pragma pack(push, 1)
 
+typedef struct _syspage_seg_t {
+	struct _syspage_seg_t *next, *prev;
 
-/* syspage_validateAddrMap type of validation bitflags */
-enum { flagValidateTop = 0x01, flagValidateMap = 0x02, flagValidateAttr = 0x04 };
+	addr_t addr;
+	size_t fileSz;
+	size_t memSz;
 
+	u32 flags;
+	u32 align;
+} syspage_seg_t;
 
-/* Initialization function */
-extern void syspage_init(void);
 
+typedef struct _syspage_program_t {
+	struct _syspage_program_t *next, *prev;
 
-/* Syspage's location functions */
+	enum { syspage_progBlob = 0, syspage_progSegs } type;
+	union {
+		struct {
+			addr_t start;
+			addr_t end;
+		} blob;
 
-extern void syspage_setAddress(addr_t addr);
+		struct {
+			syspage_seg_t *segs;
+			addr_t entryPoint;
+		} progsegs;
```

Comment:
Do we really need this?

Reply:
Currently it is used only for passing information about kernel's segments to phoenix-rtos-kernel. In the previous version, this information was crucial to set kernel's entries into specific maps.
However, the new approach introduces the concept of keeping entries in each map so the kernel has information about entries from syspage.

IMHO, it might be skipped. Phoenix-rtos-kernel will do without it. But I wasn't sure whether it is not used elsewhere in the kernel so I decided add it to be complied with out internal assumptions.

Reply:
As it is discussed above we can use only simpler program definition.